common/interpolation/include/interpolation/interpolation_utils.hpp

```diff
@@ -15,48 +15,78 @@
 #ifndef INTERPOLATION__INTERPOLATION_UTILS_HPP_
 #define INTERPOLATION__INTERPOLATION_UTILS_HPP_
 
+#include <algorithm>
 #include <array>
+#include <functional>
+#include <limits>
 #include <stdexcept>
 #include <vector>
 
 namespace interpolation_utils
 {
+constexpr double EPS = std::numeric_limits<double>::epsilon();
+
+template <class T, typename std::enable_if_t<std::is_floating_point_v<T>, bool> * = nullptr>
+bool isEqual(T a, T b)
+{
+  return abs(a - b) < std::numeric_limits<T>::epsilon();
+}
+
+// Strictly monotonic increasing
 inline bool isIncreasing(const std::vector<double> & x)
 {
   if (x.empty()) {
-    throw std::invalid_argument("Points is empty.");
+    throw std::invalid_argument("The vector x is empty.");
   }
 
-  for (size_t i = 0; i < x.size() - 1; ++i) {
-    if (x.at(i) >= x.at(i + 1)) {
-      return false;
-    }
-  }
+  // x(i) <= x(i+1) condition check
+  auto const & is_increasing_it = std::adjacent_find(x.begin(), x.end(), std::greater_equal<>());
 
-  return true;
+  return is_increasing_it == x.cend();
 }
 
+// Strictly increasing and monotonic
 inline bool isNotDecreasing(const std::vector<double> & x)
 {
   if (x.empty()) {
-    throw std::invalid_argument("Points is empty.");
+    throw std::invalid_argument("The vector x is empty.");
   }
 
-  for (size_t i = 0; i < x.size() - 1; ++i) {
-    if (x.at(i) > x.at(i + 1)) {
-      return false;
-    }
+  // x(i) < x(i+1) condition check ( x(i) == x(i+1) is allowed)
+  auto const & is_strictly_increasing_it = std::adjacent_find(x.begin(), x.end(), std::greater<>());
```

Comment:
Usually, we use `auto const` not `const auto` in autoware.universe.
Any reason to use `auto const`?

Reply:
My development environment setting like this, and this is more natural to me. Is this rule strict and compulsory? If so, I make change.